pkg/services/libraryelements/api.go

```diff
@@ -30,11 +30,31 @@ func (l *LibraryElementService) createHandler(c *models.ReqContext) response.Res
 		return response.Error(http.StatusBadRequest, "bad request data", err)
 	}
 
+	if cmd.FolderUID != nil {
+		if *cmd.FolderUID == "" {
+			cmd.FolderID = 0
+		} else {
+			folder, err := l.folderService.GetFolderByUID(c.Req.Context(), c.SignedInUser, c.OrgId, *cmd.FolderUID)
+			if err == nil && folder != nil {
+				cmd.FolderID = folder.Id
+			}
+		}
+	}
+
 	element, err := l.createLibraryElement(c.Req.Context(), c.SignedInUser, cmd)
 	if err != nil {
 		return toLibraryElementError(err, "Failed to create library element")
 	}
 
+	if element.FolderID != 0 {
+		folder, err := l.folderService.GetFolderByID(c.Req.Context(), c.SignedInUser, element.FolderID, c.OrgId)
```

Comment:
Same here, do we not want to do anything if there's an error?

Reply:
I think it shouldn't happen normally. update with an internal error instead. I don't have strong arguments/opinions here.